Create rule S6490: Arguments corresponding to the width and precision formatting options should be integers (CPP-3961) (#1453)
SonarSource/rspec: rules/S6490/cfamily/rule.adoc

> One of the features of `std::format` and related formatted functions is the possibility to specify the desired length
> and padding of the textual representation of the argument with width and precision options.

Reviewer: ```suggestion
`std::format` and the related formatting functions provide the possibility to specify the desired length
and padding of the textual representation of the argument through the width and precision options.
```

> The value of each of these options can be specified either:
> 
> *  statically, by placing the desired value in a format specification
> * dynamically, by using `{}` or `{arg-id}` to refer to a complementary function argument

Reviewer: ```suggestion
The value of each of these options can be specified:

* statically, by placing the desired value in a format specification
* or dynamically, by using `{}` or `{arg-id}` to refer to a complementary function argument.
```

> When dynamic values areused either for width or precision, an `std::format_error` exception is thrown if the corresponding argument is not built-in integer.

Reviewer: ```suggestion
When dynamic values are used either for the width or precision, an `std::format_error` exception is thrown if the corresponding argument is not a built-in integer.
```

> This rule raises an issue when the extra argument type is not an integer.

Reviewer: ```suggestion
This rule raises an issue when the argument used for the width or precision is not an integer.
```
